### ISSUE TITLE
findfeatures parameters cannot contain forward slashes

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#2474]
---

Author Name: **Jesse Mapel** (Jesse Mapel)

---

Using a forward slash in a findfeatures parameter value causes an error.  It appears to be caused by using forward slashes to separate detector, extractor, matcher, and parameters.

The following command line outputs the following error:

~~~
findfeatures match=/work/projects/messenger/Lev1/2011112/EW0211981114G.lev1.cub from=/work/projects/messenger/Lev1/2012275/EW0257563816G.lev1.cub epitolerance=1.0 ratio=0.65 hmgtolerance=2.0 networkid=EW0211981114G pointid='EW0211981114G_?????' onet=EW0211981114G.net debug=true debuglog=harshSpec_only_surf.log ALGOrithm="surf@hessianThreshold:100/surf/parameters@SaveRenderecImages:true@SavePath:/scratch/m04904/testing"
~~~
~~~
**USER ERROR** Invalid algorithm/part at or near "m04904" - too many or invalid algorithm specs detected in specification: surf@hessianThreshold:100/surf/parameters@SaveRenderecImages:true@SavePath:/scratch/m04904/testing.
~~~

**Comment by lwellerastro** [comment]
Ooh- I was just going to report this exact problem and found this post.

This would be a nice one to address if possible as I typically would findfeatures on the compute cluster and in the past have tried to specify a precise to path to place the rendered images only to have it fail as @jessemapel has reported.  The rendered images can be very useful in trying to identify what findfeatures is "seeing" to know why it may not be producing desired results.

Please let me know if Jesse's command line and data are not sufficient to reproduce the error.

**Comment by jessemapel** [comment]
Solution doesn't look too hard. Instead of splitting on `/` we can split on a regex `\/(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)` from some [googling](https://stackoverflow.com/questions/18893390/splitting-on-comma-outside-quotes). Tested it out with an [online applet](https://regex101.com/r/VLH7iZ/1) and it looks good.

**Comment by jessemapel** [comment]
This is not going to be done under 3.7.0, pushing to a future release.

**Comment by jlaura** [comment]
@lweller is going to get two images for the @students to work on

**Comment by lwellerastro** [comment]
Data to reproduce the problem can be found under /work/users/lweller/Isis3Tests/FindFeatures/SavePath/.  You will need the input images N1487301032_2_callptr.cub and N1487301209_2_callptr.cub as well as the config file image_findfeatures.conf to reproduce in your area.

To be clear, the savepath parameter will work when the specified directory is relative, but will fail when a full path is used.

This set of commands will work:

> mkdir PNGs
> 
> findfeatures algorithm=fast/brief/parameters@saverenderedimages:true@savepath:PNGs maxthreads=7 match=N1487301032_2_callptr.cub from=N1487301209_2_callptr.cub networkid=N1487301032_2 pointid='N1487301032_2_?????' onet=N1487301032_2_ff.net tolist=N1487301032_2_cubes_ff.lis description='Create image-image control network'

This will fail:

> mkdir PNGs_2
> 
> findfeatures algorithm=fast/brief/parameters@saverenderedimages:true@savepath:/work/users/lweller/Isis3Tests/FindFeatures/SavePath/PNGs_2 maxthreads=7 match=N1487301032_2_callptr.cub from=N1487301209_2_callptr.cub networkid=N1487301032_2 pointid='N1487301032_2_?????' onet=N1487301032_2_ff.net tolist=N1487301032_2_cubes_ff.lis description='Create image-image control network'

>   Group = Error
>     Program = findfeatures
>     Class   = "USER ERROR"
>     Code    = 2
>     Message = "Invalid algorithm/part at or near "users" - too many or invalid
>                algorithm specs detected in specification:
>                fast/brief/parameters@saverenderedimages:true@savepath:/work/us-
>                ers/lweller/Isis3Tests/FindFeatures/SavePath/PNGs_2"
>     File    = FeatureAlgorithmFactory.cpp
>     Line    = 605
>   End_Group

**Comment by jlaura** [comment]
This was closed by #3270
